### PR TITLE
PRO-1379 no relationship

### PR DESCRIPTION
**Pull request by abea**
Makes sure there are actual relationship fields there before we show the context menu.

### CODE REVIEW

File: modules/@apostrophecms/ui/ui/apos/components/AposSlat.vue [hunk 1]
```diff
@@ -140,6 +140,21 @@ export default {
...
       }
+    },
+    hasFields() {
+      if (!this.item._fields) {
+        return false;
+      }
+
+      let ruling = false;
+
+      for (const key in this.item._fields) {
```
Comment (abea) [inline modules/@apostrophecms/ui/ui/apos/components/AposSlat.vue:151]: I stole this like an artist.

File: modules/@apostrophecms/ui/ui/apos/components/AposSlatList.vue [hunk 2]
```diff
@@ -34,10 +34,6 @@
         />
       </transition-group>
     </draggable>
-
-    <div class="apos-slat-status">
-      {{ message }}
```
Comment (abea) [inline modules/@apostrophecms/ui/ui/apos/components/AposSlatList.vue:39]: There is no message.

File: modules/@apostrophecms/ui/ui/apos/components/AposSlatList.vue [hunk 3]
```diff
@@ -65,12 +61,6 @@ export default {
     selected: {
       type: String,
       default: null
-    },
-    field: {
```
Comment (abea) [inline modules/@apostrophecms/ui/ui/apos/components/AposSlatList.vue:69]: There is no field. This prop isn't used anymore, only serves to show the message, the message has been replaced in design, and the field is not passed in in any use of the thing.

**Review by boutell — CHANGES_REQUESTED** [review]
Hmm. I think consulting `_fields` was the original mistake here. This code would be shorter and more correct if the slat did receive the actual schema field, and could simply look at `field.schema` and check it for existence and length to determine if it is *supposed to* have a relationship. It's kind of immaterial whether it currently has populated fields or not; what matters is whether it's supposed to.

**Comment by abea** [comment]
I went with passing the boolean through because the relationship input is the relevant context and we don't need to make the slats do work looking for stuff.

**Review by boutell — APPROVED** [review]

**Pull request merged**